addons/google_calendar/models/calendar.py

```diff
@@ -209,6 +216,126 @@ def _odoo_reminders_commands(self, reminders=()):
+            rec_fields = list(self._get_recurrent_fields())
+            [rec_vals] = base_event.read(rec_fields)
+
+            # Get information from the current recurrence.
+            old_recurrence = self.env['calendar.recurrence']
+            old_recurrence |= self.recurrence_id
+            old_recurrence_values = {field: rec_vals.pop(field) for field in rec_fields if field in rec_vals}
+            linked_events = self.env['calendar.event']
+            linked_events |= old_recurrence.calendar_event_ids
+
+            if recurrence_values:
+                # Get start date from parameters (or base_event) and update recurrence.
+                start_date = time_values['start'].date() if 'start' in time_values else base_event.start.date()
+                new_values = {
+                    **old_recurrence_values,
+                    **base_event._get_recurrence_params_by_date(start_date),
+                    **recurrence_values,
+                }
+                new_values.pop('rrule')
+                old_recurrence.write(new_values)
+
+            # Optimization: if there are recurrence_values, the base_event won't be archived.
+            events_to_patch = linked_events if not recurrence_values else base_event
+
+            # Update events information (time, description, etc).
+            for record in events_to_patch:
+                record.need_sync = False
+                # Only time updates within the same day are allowed: combine new specified time with current date.
+                if 'start' in time_values:
+                    time_values['start'] = datetime.combine(record.start, time_values['start'].time())
+
+                if 'stop' in time_values:
+                    time_values['stop'] = datetime.combine(record.stop, time_values['stop'].time())
+                record.write({**values, **time_values, **{'need_sync': False}})
+
+            # Update the recurrence structure and reconstruct events except for the base_event.
+            if recurrence_values:
+                old_recurrence.calendar_event_ids = [(4, base_event.id)]
+                (linked_events - base_event).write({'need_sync': False, 'active': False})
+                old_recurrence._apply_recurrence()
+
+    def action_mass_archive(self, recurrence_update_setting):
+        """
+        The aim of this action is to be called from sync calendar module when mass deletion is not possible.
+        Single request archiving in Google Calendar API: no spam is sent and outside users are also notified.
+        """
+        self.ensure_one()
+        google_service = GoogleCalendarService(self.env['google.service'])
+        if recurrence_update_setting == 'all_events' or self == self.recurrence_id.base_event_id:
+            self.recurrence_id._google_delete(google_service, self.recurrence_id.google_id, is_recurrence=True)
+            self.recurrence_id.calendar_event_ids.write({'active': False, 'need_sync': False})
```

Comment:
{'active': False, 'need_sync': False} is regularly used and {'active': False} is also used a lot in calendar.
Could we define a method?

```py
@api.model
def _get_archive_values(self):
    return {'active': False}
```
and in google_calendar, it wold return:

```py
def _get_archive_values(self):
    archive_values = super()._get_archive_values()
    return {**archive_values, 'need_sync': False} 
```

It would allow you to call super and you would have to make sure self.recurrence_id._google_delete(google_service, self.recurrence_id.google_id, is_recurrence=True) is called before calling the equivalent of self.recurrence_id.calendar_event_ids.write({'active': False, 'need_sync': False}).
What do you think ?